src/app/campaign/store/reducers/flight.reducer.ts

```diff
@@ -146,6 +147,11 @@ export function reducer(state = initialState, action: CampaignActions): State {
       }
       return newState;
     }
+    case ActionTypes.CAMPAIGN_FLIGHT_PREVIEW_CREATE_SUCCESS: {
+      const { flight, status, statusMessage } = action.payload;
```

Comment:
I started out patching the entire previewed flight-doc back onto `localFlight`.  But that caused all kinds of crazy form flickering ... especially if you're changing fields fast and generating multiple in-progress preview POSTs.  Each time a 201 comes back, it resets your form back to that state!

So I went for the minimal status/message patching here.  Let me know what you think of that!

Reply:
ugh, yeah I could see that being gross. But doesn't this also cause a state cycle update? Because even if you weren't updating the localFlight field, the whole state should be replaced on each update to the state, so do the selectors see this as a change? I'm not sure what the difference would be where you're seeing that flickering, seems like it would do the same thing with this.

I'm wondering if the status and statusMessage should be stored on the reducer entity FlightState instead of the flight model? I understand that you'd then have to pass it down to the inventory.component separate from the Flight Input, but previewError has its own selector and gets passed in there.

Reply:
The difference is that the other `localFlight` fields are in the form, and status/message are not.  So with something like `totalGoal` -

1. Change `totalGoal = 10` in the form
2. Fires off POST preview
3. Change `totalGoal = 11` in the form
4. Fires off a 2nd POST preview
5. Preview (2) returns, changes form `totalGoal` back to 10.  _(I don't _think_ this fires off a 3rd POST preview - something is at least preventing that)_
6. Preview (4) returns, changes form `totalGoal` back to 11.

Reply:
I can see the flickering with that, ok. But hmm, you know it seems like what you describe in itself is a problem. Not your problem in this PR (actually my fault,) but we should be using `switchMap` in the `createFlightPreview$` @Effect instead of `mergeMap`. It should cancel the prior request with totalGoal of 10 and only respond with 11.

Reply:
Oh, cool!  I didn't even know that was a thing, but sounds like a good solution.  Thanks ngrx!